ignite/metrics/metric.py

```diff
@@ -546,6 +550,41 @@ def is_attached(self, engine: Engine, usage: Union[str, MetricUsage] = EpochWise
         usage = self._check_usage(usage)
         return engine.has_event_handler(self.completed, usage.COMPLETED)
 
+    def state_dict(self) -> OrderedDict:
+        """Method returns state dict with attributes of the metric specified in its
+        `_state_dict_all_req_keys` attribute. Can be used to save internal state of the class.
+
+        If there's an active distributed configuration, some collective operations is done and
+        the list of values across ranks is saved under each attribute's name in the dict.
+        """
+        state = OrderedDict()
+        for attr_name in self._state_dict_all_req_keys:
+            attr = getattr(self, attr_name)
+            if idist.get_world_size() == 1:
+                state[attr_name] = [attr]
+            else:
+                if isinstance(attr, (float, torch.Tensor)):
+                    state[attr_name] = cast(List[Any], idist.all_gather(attr))
+                else:
+                    state[attr_name] = [None] * idist.get_world_size()
+                    torch_dist.all_gather_object(state[attr_name], attr)
```

Comment:
Let's raise an error here now. We can add a support for picklable objects later with `idist.all_gather`.